src/jvm/main/org/jetbrains/kotlinx/lincheck/paramgen/DoubleGen.kt

```diff
@@ -0,0 +1,69 @@
+/*
+* #%L
+* Lincheck
+* %%
+* Copyright (C) 2015 - 2018 Devexperts, LLC
+* %%
+* This program is free software: you can redistribute it and/or modify
+* it under the terms of the GNU Lesser General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Lesser Public License for more details.
+*
+* You should have received a copy of the GNU General Lesser Public
+* License along with this program.  If not, see
+* <http://www.gnu.org/licenses/lgpl-3.0.html>.
+* #L%
+*/
+package org.jetbrains.kotlinx.lincheck.paramgen
+
+import java.util.*
+
+class DoubleGen(configuration: String) : ParameterGenerator<Double> {
+    private val random = Random(0)
+    private var begin = 0.0
+    private var end = 0.0
+    private var step = 0.0
+    override fun generate(): Double {
+        val delta = end - begin
+        if (step == 0.0) // step is not defined
+            return begin + delta * random.nextDouble()
+        val maxSteps = (delta / step).toInt()
+        return begin + delta * random.nextInt(maxSteps + 1)
+    }
+
+    companion object {
```

Comment:
You don't need this companion object, just put these constants right after the class declaration, on the top-level.